src/repositories/ImmutableRepository.ts

```diff
@@ -0,0 +1,124 @@
+import {
+  BaseEntity,
+  DeepPartial,
+  EntityManager,
+  EntityTarget,
+  FindOneOptions,
+  FindOptionsRelations,
+  FindOptionsWhere,
+} from 'typeorm';
+import { QueryDeepPartialEntity } from 'typeorm/query-builder/QueryPartialEntity.js';
+
+import { AbstractRepository } from './AbstractRepository';
+import {
+  EntryNotFoundAfterInsertException,
+  IllegalArgumentException,
+  InsertionException,
+} from './errors';
+
+type CreateBody<T extends BaseEntity> = DeepPartial<T>;
+
+/**
+ * Abstract class representing a repository for immutable entities.
+ * Extends the AbstractRepository class to provide additional functionality for adding and retrieving entities.
+ *
+ * @template T The type of the entity.
+ */
+export abstract class ImmutableRepository<T extends BaseEntity> extends AbstractRepository<T> {
+  /** The primary key of the entity used during the find. */
+  protected readonly primaryKey: keyof T;
```

Comment:
This `primaryKey` variable's name is already used in functions, I feel like it can be confusing. What about `primaryKeyName` ? Otherwise, do you have other ideas ?